tests/fixtures/optimizer/qualify_tables.sql

```diff
@@ -19,25 +19,46 @@ SELECT (SELECT y.c FROM c.db.y AS y) FROM c.db.x AS x;
 SELECT * FROM x PIVOT (SUM(a) FOR b IN ('a', 'b'));
 SELECT * FROM c.db.x AS x PIVOT(SUM(a) FOR b IN ('a', 'b')) AS _q_0;
 
-----------------------------
--- Expand join constructs
-----------------------------
+----------------------------------------------------------
+-- Unnest wrapped tables / joins, expand join constructs
+----------------------------------------------------------
 
--- This is valid in Trino, so we treat the (tbl AS tbl) as a "join construct" per postgres' terminology.
-SELECT * FROM (tbl AS tbl) AS _q_0;
-SELECT * FROM (SELECT * FROM c.db.tbl AS tbl) AS _q_0;
+# title: redundant parentheses (1)
+SELECT * FROM (tbl AS tbl);
+SELECT * FROM c.db.tbl AS tbl;
+
+# title: redundant parentheses (2)
+SELECT * FROM ((((tbl AS tbl))));
+SELECT * FROM c.db.tbl AS tbl;
+
+# title: redundant parentheses around join construct
+SELECT * FROM a LEFT JOIN (b INNER JOIN c ON c.id = b.id) ON b.id = a.id;
```

Comment:
do we also handle

select from ( (b join c) join d)

so join d is not on a table, but on a paren / subquery?

Reply:
Nope, parser error, good catch! Working on a fix

Reply:
can you check if it's valids ql?

Reply:
Already did, it runs on postgres:

```
georgesittas=# with b as (select 1), c as (select 1), d as (select 1) select * from ( (b cross join c) cross join d);
 ?column? | ?column? | ?column?
----------+----------+----------
        1 |        1 |        1
(1 row)
```

Reply:
Tried to resolve this in https://github.com/tobymao/sqlglot/pull/1894/commits/a481a6c62de037f7000470b81156bf90b3bdff7f